lib/Sema/CSDiag.cpp

```diff
@@ -2273,6 +2273,27 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
       } else {
         // Otherwise the static member lookup was invalid because it was
         // called on an instance
+        
+        // Handle enum element lookup on instance type
+        auto lookThroughBaseObjTy = baseObjTy->lookThroughAllAnyOptionalTypes();
+        if (lookThroughBaseObjTy->is<EnumType>()
+            || lookThroughBaseObjTy->is<BoundGenericEnumType>()) {
+          bool triedToAccessEnumElement = false;
+          for (auto cand : result.UnviableCandidates) {
+            ValueDecl *decl = cand.first;
+            if (isa<EnumElementDecl>(decl)) {
+              triedToAccessEnumElement = true;
+              break;
+            }
+          }
+          if (triedToAccessEnumElement) {
+            diagnose(loc, diag::could_not_use_enum_element_on_instance,
```

Comment:
I tried adding a fix-it but there wasn’t enough context available to decide whether a leading dot would do (at least as far as I could see). There was also a reason was difficult/impossible as well (I’m just having a look to remember again). Since the fully qualified type was often far more verbose than it really needed to be, so I decided against it.